### PR TITLE
allow customizing the worker names and do not override defaults

### PR DESCRIPTION
**Pull request by jglaser**
In order to support using `dask-mpi` as a backend for multiple workers per job in `dask-gateway`, I am introducing the capability to set a custom name, to which the rank is appended. This change is needed for keeping track of workers in `dask-gateway` (PR to be submitted). In addition, I am reverting some of the default overrides for the CLI options, as those for example set the TCP protocol and a scheduler-file, whereas the `dask-gateway` use case requires being able to specify a scheduler address from the command line.

### CODE REVIEW

**Review by kmpaul — CHANGES_REQUESTED** [review]
@jglaser:  So, this PR does more than it says in the title.  It removes some default values for many parameters (causing failures in the CI), in addition to allowing the user to specify base worker names.  Maybe these two changes could be split into separate PRs?  I'm positive the worker name changes can be done very quickly.  The other changes will require some debugging, I think.

**Comment by kmpaul** [comment]
@jglaser:  I suspect that the

File: dask_mpi/cli.py [hunk 1]
```diff
@@ -9,25 +9,30 @@
 
 
 @click.command()
+@click.argument("scheduler_address",
+    type=str,
+    required=False
+)
+
```
Comment (kmpaul) [inline dask_mpi/cli.py:16]:
```suggestion
@click.argument("scheduler_address", type=str, required=False)
```

File: dask_mpi/cli.py [hunk 2]
```diff
@@ -9,25 +9,30 @@
...
 )
 @click.option(
     "--interface", type=str, default=None, help="Network interface like 'eth0' or 'ib0'"
 )
 @click.option(
-    "--protocol", type=str, default="tcp", help="Network protocol to use like TCP"
+    "--protocol", type=str, default=None, help="Network protocol to use like TCP"
 )
-@click.option("--nthreads", type=int, default=1, help="Number of threads per worker.")
+@click.option("--nthreads", type=int, default=None, help="Number of threads per worker.")
```
Comment (kmpaul) [inline dask_mpi/cli.py:35]:
```suggestion
@click.option(
    "--nthreads", type=int, default=None, help="Number of threads per worker."
)
```

File: dask_mpi/cli.py [hunk 3]
```diff
@@ -83,6 +95,7 @@ def main(
     worker_options,
     scheduler_port,
     protocol,
+    name
```
Comment (kmpaul) [inline dask_mpi/cli.py:98]:
```suggestion
    name,
```

**Review by kmpaul — CHANGES_REQUESTED** [review]
These suggestions should fix your linting errors.

**Comment by kmpaul** [comment]
@jglaser:  Hey!  I got this to pass.  Do you want me to merge this in now?

**Review by kmpaul — APPROVED** [review]
Got it to pass!

**Comment by jglaser** [comment]
Thanks for fixing it , @kmpaul .... and sorry for abandoning the issue. Please feel free to merge.

**Comment by kmpaul** [comment]
Now worries.  And thanks, @jglaser!

**Pull request merged**